apps/system/js/app_usage_metrics.js

```diff
@@ -850,6 +895,24 @@
         }
       }
     }
+
+    // loop through all the search providers that we have data for
+    // and merge the new searches into the old searches.
+    for (var provider in newbatch.data.searches) {
+      var newsearch = newbatch.data.searches[provider];
+      var oldsearch = this.data.searches[provider];
+
+      for (var daykey in newsearch) {
+        var daySearch = oldsearch[daykey];
```

Comment:
Yes, this is a good catch.  Thank you.  I've added an initialization if it's null.